metrics/types.go

```diff
@@ -65,3 +65,16 @@ func (t packetType) String() string {
 		panic("unknown packet type")
 	}
 }
+
+type timeoutReason logging.TimeoutReason
+
+func (r timeoutReason) String() string {
+	switch logging.TimeoutReason(r) {
+	case logging.TimeoutReasonHandshake:
+		return "handshake_timeout"
+	case logging.TimeoutReasonIdle:
+		return "idle_timeout"
+	default:
+		panic("unknown timeout reason")
```

Comment:
Don't panic, just return `"unknown"`. Metrics is never a good enough reason to crash a program.

Reply:
I tend to agree, although if we don't panic, those bugs will probably go unnoticed for a long time (until someone looks at the logs and sees a bunch of `unknown`s there). Anyway, this is a larger change, since we currently panic both in the `logging` and in the `qlog` package.